backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadService.java

```diff
@@ -114,48 +126,77 @@ public PatientBulkUploadResponse processPersonCSV(InputStream csvStream, UUID fa
           continue;
         }
 
-        _personService.addPatient(
-            facilityId,
-            null, // lookupID
-            extractedData.getFirstName().getValue(),
-            extractedData.getMiddleName().getValue(),
-            extractedData.getLastName().getValue(),
-            extractedData.getSuffix().getValue(),
-            parseUserShortDate(extractedData.getDateOfBirth().getValue()),
-            address,
-            country,
-            List.of(
-                new PhoneNumber(
-                    parsePhoneType(extractedData.getPhoneNumberType().getValue()),
-                    extractedData.getPhoneNumber().getValue())),
-            parsePersonRole(extractedData.getRole().getValue(), false),
-            List.of(extractedData.getEmail().getValue()),
-            convertRaceToDatabaseValue(extractedData.getRace().getValue()),
-            convertEthnicityToDatabaseValue(extractedData.getEthnicity().getValue()),
-            null,
-            convertSexToDatabaseValue(extractedData.getBiologicalSex().getValue()),
-            parseYesNo(extractedData.getResidentCongregateSetting().getValue()),
-            parseYesNo(extractedData.getEmployedInHealthcare().getValue()),
-            null,
-            null);
+        // create new person
+        // - pass in the organization they belong to, as saved up above
+        // - set facility on person
+        // add person to person list
+        // create List of Phone numbers for that person following PersonService steps
+        // add phone numbers to phone numbers list
+        // exit while loop
+        // do the save for persons and phone numbers
+        // see how much better it is
+
+        // create new person with current organization, then add to new patients list
+        Person newPatient =
+            new Person(
+                currentOrganization,
+                null,
+                extractedData.getFirstName().getValue(),
+                extractedData.getMiddleName().getValue(),
+                extractedData.getLastName().getValue(),
+                extractedData.getSuffix().getValue(),
+                parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+                address,
+                country,
+                parsePersonRole(extractedData.getRole().getValue(), false),
+                List.of(extractedData.getEmail().getValue()),
+                convertRaceToDatabaseValue(extractedData.getRace().getValue()),
+                convertEthnicityToDatabaseValue(extractedData.getEthnicity().getValue()),
+                null,
+                convertSexToDatabaseValue(extractedData.getBiologicalSex().getValue()),
+                parseYesNo(extractedData.getResidentCongregateSetting().getValue()),
+                parseYesNo(extractedData.getEmployedInHealthcare().getValue()),
+                null,
+                null);
+        newPatient.setFacility(assignedFacility); // might be null, that's fine
```

Comment:
that's a good question, maybe @emmastephenson would know why we don't have a constructor with facility? If there are no concerns, I can add a new constructor.

Reply:
I have two theories here, but I don't know which is the real root cause:
1. We didn't originally have both facilities and organizations, just organizations, and this constructor is a reflection of that
2. We still allow all-facility patient records, so createPerson().setFacility() seems valid there too.

That said, I'm inclined to agree here - I think the neatest solution is probably to create a new constructor in the `Person` model and include facility there.

Reply:
@emmastephenson That makes sense. And in the case that `all facilities` is selected, it'll be null and that's fine.

Reply:
@emmastephenson @BobanL can you take a look at the new impl and let me know if it makes sense to you? I ended up using `Optional<Facility>` so it would keep things in the `PatientBulkUploadService` cleaner. I think this also handles the null case nicely.

Reply:
https://github.com/CDCgov/prime-simplereport/blob/772af8cf3c097024c60cff573f76e8fb339ee4d9/backend/src/main/java/gov/cdc/usds/simplereport/db/model/Person.java#L179-L179